include/oneapi/dpl/internal/async_impl/glue_async_impl.h

```diff
@@ -0,0 +1,163 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2017-2021 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_GLUE_ASYNC_IMPL_H
+#define _ONEDPL_GLUE_ASYNC_IMPL_H
+
+#if _ONEDPL_HETERO_BACKEND
+#    include "async_impl_hetero.h"
+#endif
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace experimental
+{
+
+// [async.transform]
+template <class _ExecutionPolicy, class _ForwardIterator1, class _ForwardIterator2, class _UnaryOperation,
+          class... _Events>
+oneapi::dpl::__internal::__enable_if_async_execution_policy<
+    _ExecutionPolicy, oneapi::dpl::__internal::__future<_ForwardIterator2>, _Events...>
+transform_async(_ExecutionPolicy&& __exec, _ForwardIterator1 __first, _ForwardIterator1 __last,
+                _ForwardIterator2 __result, _UnaryOperation __op, _Events&&... __dependencies)
+{
+    wait_for_all(__dependencies...);
```

Comment:
`wait_for_all` takes `const` lvalue reference but here and in other places you pass forwarding reference without `std::forward`. I just don't understand what's the value from forwarding reference then